src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -138,4 +146,41 @@ private void cleanTags(Map<String, String> tags) {
         .retryWhen(appProperties.getRetryInsertDownsampled().build())
         .checkpoint();
   }
+
+  private Mono<?> storeMetricGroup(Metric metric) {
+    String metricGroup = metric.getMetric();
+    String metricName = metric.getMetric();
+    if(metric.getMetric().contains("_"))  {
+      String[] strArr = metric.getMetric().split("_",2);
+      metricGroup = strArr[0];
+      metricName = strArr[1];
+    }
+
+    MetricGroupAndName metricGroupAndName = new MetricGroupAndName()
```

Comment:
Move this to line 151 and remove the variables `metricGroup` and `metricName` and simply update the values of `metricGroupName` inside the if condition.